### DISCUSSION TITLE
Need to Select an automation tool between Cypress and Selenium Webdriver?

### DISCUSSION THREAD

**Discussion by gazalshukla** [cypress-io/cypress#14454] [Questions and Help]
0

So I am going to work on a project which requires end-to-end Automation. My team prefers using Cypress.

Requirement: We want to build our automation framework which tests chrome, safari, Edge and Firefox browsers on desktop, mobile and tablet.

For Selenium : Broswerstack is there as solution to test chrome, safari, Edge and Firefox browsers on desktop, mobile and tablet.

For Cypress : Cypress doesn't support Safari ( as far as i know) . I have tried Applitools but they also dont give results for mobile and tablet .

Is there any way possible for Cypress? If yes, Please suggest me solutions.

**Comment by bahmutov** [comment]
We are currently evaluating adding support for desktop Safari browser, but there are technical challenges there. Keep an eye on https://github.com/cypress-io/cypress/issues/6422

**Reply by gazalshukla** [reply]
And is any there any solution for running test cases in mobile or tablets for cypress ?
Like for selenium, we have Browser stack.

**Comment by bahmutov** [comment]
Unfortunately there is no way to run The test on a browser running on a mobile device or a tablet
